code/obj/item/sec_tape.dm

```diff
@@ -0,0 +1,162 @@
+#define MAXTAPE 40
+#define STARTAMOUNT 1
+/// The security tape roll object, used to create security tapes.
+/obj/item/sec_tape
+	name = "security tape roll"
+	var/base_name = "security tape roll"
+	desc = "A roll of security tape."
+	amount = STARTAMOUNT
+	max_stack = MAXTAPE
+	stack_type = /obj/item/sec_tape
+	icon = 'icons/obj/sec_tape.dmi'
+	icon_state = "sec_tape_roll"
+	throwforce = 2
+	w_class = W_CLASS_TINY
+	throw_speed = 2
+	throw_range = 5
+	flags = TABLEPASS|EXTRADELAY|FPRINT|CONDUCT|ONBELT
+	stamina_damage = 5
+	stamina_cost = 5
+	stamina_crit_chance = 10
+	rand_pos = 1
+	event_handler_flags = USE_GRAB_CHOKE | USE_FLUID_ENTER
+	special_grab = /obj/item/grab
+	inventory_counter_enabled = 1
+
+	New(loc, length = STARTAMOUNT)
+		src.amount = length
+		pixel_x = rand(-2,2)
+		pixel_y = rand(-2,2)
+		..(loc)
+		BLOCK_SETUP(BLOCK_ROPE)
+
+	before_stack(atom/movable/O as obj, mob/user as mob)
+		user.visible_message("<span class='notice'>[user] begins awkwardly sticking the tape back onto the roll.</span>")
+
+	after_stack(atom/movable/O as obj, mob/user as mob, var/added)
+		boutput(user, "<span class='notice'>You finish sticking the tape back together.</span>")
+
+	custom_suicide = 1
+	suicide(var/mob/user as mob)
+		if (!src.user_can_suicide(user))
+			return 0
+		user.visible_message("<span class='alert'><b>[user] wraps some tape around \his neck and tightens it.</b></span>")
+		user.take_oxygen_deprivation(160)
+		SPAWN(50 SECONDS)
+			if (user && !isdead(user))
+				user.suiciding = 0
+		return 1
+
+	proc/use(var/used)
+		if (src.amount < used)
+			return 0
+		else if (src.amount == used)
+			qdel(src)
+			return 1
+		else
+			amount -= used
+			tooltip_rebuild = 1
+			src.UpdateIcon()
+			return 1
+
+	proc/take(var/amt, var/newloc)
+		if (amt > amount)
+			amt = amount
+			tooltip_rebuild = 1
+			src.UpdateIcon()
+		if (amt == amount)
+			if (ismob(loc))
+				var/mob/owner = loc
+				owner.u_equip(src)
+			set_loc(newloc)
+			return src
+		src.use(amt)
+		var/obj/item/sec_tape/C = new /obj/item/sec_tape(newloc)
+		C.amount = amt
+		return
+
+	update_icon()
+		inventory_counter?.update_number(amount)
+		return
+
+/obj/item/sec_tape/vended //Tape in secvends has a pre-set amount of uses
+	New(loc, length)
+		..(loc, 20)
+
+/obj/item/sec_tape/get_desc()
+	return " There's [amount] length[s_es(amount)] left."
```

Comment:
pretty sure you don't need the leading space here

Reply:
Removed